File: Structural/Bridge/bridge.py
from abc import ABC, abstractmethod

class Implementor(ABC):
    @abstractmethod
    def operation_impl(self) -> None:
        pass

class ConcreteImplementorA(Implementor):
    def operation_impl(self) -> None:
        print("ConcreteImplementorA Operation")

class ConcreteImplementorB(Implementor):
    def operation_impl(self) -> None:
        print("ConcreteImplementorB Operation")

class Abstraction(ABC):
    def __init__(self, implementor: Implementor):
        self.implementor = implementor

    @abstractmethod
    def operation(self) -> None:
        pass

class RefinedAbstraction(Abstraction):
    def operation(self) -> None:
        print("RefinedAbstraction: ", end="")
        self.implementor.operation_impl()

File: Structural/Bridge/bridge.cs
public interface IImplementor {
    void OperationImpl();
}

public class ConcreteImplementorA : IImplementor {
    public void OperationImpl() {
        Console.WriteLine("ConcreteImplementorA Operation");
    }
}

public class ConcreteImplementorB : IImplementor {
    public void OperationImpl() {
        Console.WriteLine("ConcreteImplementorB Operation");
    }
}

public abstract class Abstraction {
    protected IImplementor implementor;
    public Abstraction(IImplementor implementor) {
        this.implementor = implementor;
    }
    public abstract void Operation();
}

public class RefinedAbstraction : Abstraction {
    public RefinedAbstraction(IImplementor implementor) : base(implementor) {}
    public override void Operation() {
        Console.Write("RefinedAbstraction: ");
        implementor.OperationImpl();
    }
}